buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -232,6 +232,60 @@ class BuildPlugin implements Plugin<Project> {
         project.ext.java9Home = project.rootProject.ext.java9Home
     }
 
+    static void requireDocker(final Task task) {
+        final Project rootProject = task.project.rootProject
+        if (rootProject.hasProperty('requiresDocker') == false) {
+            rootProject.rootProject.ext.requiresDocker = []
+            rootProject.gradle.taskGraph.whenReady { TaskExecutionGraph taskGraph ->
+                final String dockerBinary
+                if (new File('/usr/bin/docker').exists()) {
+                    dockerBinary = '/usr/bin/docker'
+                } else if (new File('/usr/local/bin/docker').exists()) {
+                    dockerBinary = '/usr/local/bin/docker'
+                }
+                final int exitCode
+                final String dockerErrorOutput
+                if (dockerBinary == null) {
+                    exitCode = -1
+                    dockerErrorOutput = null
+                } else {
+                    final Process process = new ProcessBuilder(dockerBinary, "images").start()
+                    exitCode = process.waitFor()
+                    if (exitCode == 0) {
+                        return
+                    }
+                    final Scanner scanner = new Scanner(process.errorStream).useDelimiter("\\A")
+                    dockerErrorOutput = scanner.hasNext() ? scanner.next() : ""
+                    process.closeStreams()
+                }
+                final List<String> tasks =
+                        ((List<Task>)rootProject.requiresDocker).findAll { taskGraph.hasTask(it) }.collect { "  ${it.path}".toString()}
+                if (tasks.isEmpty() == false) {
+                    if (dockerBinary == null) {
+                        throw new GradleException(
+                                String.format(
+                                        Locale.ROOT,
+                                        "Docker is required to run the following task%s: \n%s",
+                                        tasks.size() > 1 ? "s" : "",
+                                        tasks.join('\n')))
+                    } else {
+                        assert exitCode > 0 && dockerErrorOutput != null
+                        throw new GradleException(
+                                String.format(
```

Comment:
I think it would be useful to clearly distinguish docker not being found in one of the two expected locations, vs an error when trying to run the binary.

Reply:
@rjernst So I think I did distinguished the two cases, or am I misunderstanding what you mean?

Reply:
Sorry, you are right. My eyes glossed over the first if block.

Reply:
If you use a task property and a `tasks.all` as I suggested, I think this can all be moved together so that the if/elseif on the docker file check can end with an else that throws the error, so you don't need to fake out the exit code.